core/sr-api-macros/src/decl_runtime_apis.rs

```diff
@@ -57,9 +57,21 @@ const CHANGED_IN_ATTRIBUTE: &str = "changed_in";
 ///
 /// Is used when a trait method was renamed.
 const RENAMED_ATTRIBUTE: &str = "renamed";
+/// The `skip_initialize_block` attribute.
+///
+/// Is used when a trait method does not require that the block is initialized
```

Comment:
don't all of them?

Reply:
No, `initialize_block` itself does not need to have the block initialized and `execute_block`.